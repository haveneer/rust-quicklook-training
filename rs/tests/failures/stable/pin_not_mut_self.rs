use std::marker::PhantomPinned;
use std::pin::pin;

struct MyStruct {
    value: i32,
    _pin: PhantomPinned, // Make it `!Unpin`
}

impl MyStruct {
    fn set_value(&mut self, new_value: i32) {
        self.value = new_value;
    }
}

fn main() {
    let pinned: Pin<&mut MyStruct> = pin!(MyStruct {
        value: 10,
        _pin: PhantomPinned
    });

    pinned.set_value(0);

    // let mut other_value = MyStruct { value: 3, _pin: PhantomPinned };
    // std::mem::swap(&mut other_value, pinned.get_mut());
}
