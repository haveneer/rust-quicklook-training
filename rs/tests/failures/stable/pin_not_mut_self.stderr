error[E0412]: cannot find type `Pin` in this scope
  --> tests/failures/stable/pin_not_mut_self.rs:16:17
   |
16 |     let pinned: Pin<&mut MyStruct> = pin!(MyStruct {
   |                 ^^^ not found in this scope
   |
help: consider importing this struct
   |
1  + use std::pin::Pin;
   |
